pool/paymentmgr.go

```diff
@@ -148,6 +166,21 @@ func NewPaymentMgr(pCfg *PaymentMgrConfig) (*PaymentMgr, error) {
 	return pm, nil
 }
 
+// isProcessing returns whether the payment manager is in the process of
+// paying out dividends.
+func (pm *PaymentMgr) isProcessing() bool {
+	pm.mtx.Lock()
+	defer pm.mtx.Unlock()
+	return pm.processing
+}
+
+// setProcessing sets the processing flag to the provided boolean.
+func (pm *PaymentMgr) setProcessing(status bool) {
+	pm.mtx.Lock()
+	pm.processing = status
```

Comment:
Using a Mutex to protect a bool like this seems unnecessary. The mutex itself can be used to wait until any on-going processing has finished.

Reply:
The plan is to have any subsequent calls while the flag is set terminate immediately, not wait to run in the future.